elf/elf.go

```diff
@@ -546,19 +548,19 @@ func (b *Module) Load() error {
 
 func (b *Module) initializePerfMaps() error {
 	for name, m := range b.maps {
-		var cpu C.int = 0
-
 		if m.m != nil && m.m.def._type != C.BPF_MAP_TYPE_PERF_EVENT_ARRAY {
 			continue
 		}
 
-		for {
-			pmuFD, err := C.perf_event_open_map(-1 /* pid */, cpu /* cpu */, -1 /* group_fd */, C.PERF_FLAG_FD_CLOEXEC)
+		cpus, err := utils.GetOnlineCPUs()
+		if err != nil {
+			return fmt.Errorf("failed to determine online cpus: %v", err)
+		}
+
+		for _, cpu := range cpus {
```

Comment:
Here, `cpu` is a `uint`. It used to be a `C.int` before the patch.

Do we have the guarantee that these 2 types have the same size? It matters because of `unsafe.Pointer(&cpu)` below. If it is not guaranteed, we might need to assign to a temporary variable before using its address.